internal/publish.go

```diff
@@ -171,6 +172,11 @@ func createTgz(composeContent []byte, appDir string) ([]byte, error) {
 				return fmt.Errorf("Tar: Can't tar non regular types yet: %s", header.Name)
 			}
 		}
+		// reset the file's timestamps, otherwise hashes of the resultant TGZs will differ
+		// even if their content is the same
+		header.ChangeTime = time.Time{}
+		header.AccessTime = time.Time{}
+		header.ModTime = time.Time{}
```

Comment:
should we use `fi.ModTime` so that the timestamp is a little more accurate? I assume that value will be consistent (git would set it?)?

Reply:
What for? Also, I think, `fi.ModTime` is equal to the time when `git clone` was done by CI workers, which means that hashes for the same App but different commits/CI jobs will differ.

Reply:
you are correct. i had no idea it worked that way. i wonder how docker is producing consistent tarballs. doesn't really matter i guess as the timestamp shouldn't be critical and this keeps things concise. If no time is set, i guess the extracted files all have epoch as their timestamp?

Reply:
> If no time is set, i guess the extracted files all have epoch as their timestamp?
yes.

> i wonder how docker is producing consistent tarballs.

The docker tarball contains files produced by the docker build procedure and the layers, effectively tarballs, are stored in the cache, hence the overall image is reproducible if nothing is rebuilt.